var/spack/repos/builtin/packages/protobuf/package.py

```diff
@@ -46,6 +48,25 @@ class Protobuf(CMakePackage):
     # first fixed in 3.4.0: https://github.com/google/protobuf/pull/3406
     patch('pkgconfig.patch', when='@:3.3.2')
 
+    def fetch_remote_versions(self):
+        """Ignore additional source artifacts uploaded with releases,
+           only keep known versions
+           fix for https://github.com/LLNL/spack/issues/5356"""
+        return dict(map(lambda u:
```

Comment:
Although flake is OK with this, IMO it could be more-clearly formatted:

```
return dict(map(
  lambda u: (u, self.url_for_version(u)),
  spack.util.web.find_versions_of_archive(
    self.all_urls, self.list_url, self.list_depth)
))
```

Other than that your changes LGTM. No need to fight flake though if it's hard to format it in a matter that flake accepts. I'll get another chance to check this in a few hours. If you think the formatting is fine let me know.

Reply:
no problem, will adjust to your recommended line-breaking